src/benchmark/operators/table_scan_sorted_benchmark.cpp

```diff
@@ -0,0 +1,277 @@
+  std::shared_ptr<Table> table;
+  std::shared_ptr<TableWrapper> table_wrapper;
+
+  table = std::make_shared<Table>(*table_column_definitions, TableType::Data);
+
+  if (!order_by.has_value()) {
+    auto values = std::vector<std::string>(table_size);
+    for (int i = 0; i < table_size; i++) {
+      auto str = std::to_string(i);
+      str = std::string(string_length - str.length(), '0').append(str);
+      values[i] = str;
+    }
+    std::random_shuffle(values.begin(), values.end());
+    for (const auto& str : values) {
+      table->append({str});
+    }
+  } else if (order_by.value() == OrderByMode::Ascending || order_by.value() == OrderByMode::AscendingNullsLast) {
+    for (int i = 0; i < table_size; i++) {
+      auto str = std::to_string(i);
+      str = std::string(string_length - str.length(), '0').append(str);
+      table->append({str});
+    }
+  } else {
+    for (int i = 0; i < table_size; i++) {
+      auto str = std::to_string(table_size - i - 1);
+      str = std::string(string_length - str.length(), '0').append(str);
+      table->append({str});
+    }
+  }
+
+  if (encoding_type != EncodingType::Unencoded) {
+    ChunkEncoder::encode_all_chunks(table, SegmentEncodingSpec(encoding_type));
+  }
+
+  if (order_by.has_value()) {
+    for (auto& chunk : table->chunks()) {
+      chunk->get_segment(ColumnID(0))->set_sort_order(order_by.value());
+    }
+  }
+
+  table_wrapper = std::make_shared<TableWrapper>(std::move(table));
+  table_wrapper->execute();
+
+  return table_wrapper;
+}
+
+void BM_TableScanSorted(
+    benchmark::State& state, const int table_size, const double selectivity,
+    const PredicateCondition predicate_condition, const EncodingType encoding_type,
+    const std::optional<OrderByMode> order_by,
+    std::function<std::shared_ptr<TableWrapper>(const int, const EncodingType, const OrderByMode)> table_creator) {
+  _clear_cache();
+
+  AllTypeVariant search_value;
+  switch (predicate_condition) {
+    case PredicateCondition::LessThanEquals:
+    case PredicateCondition::LessThan:
+      search_value = static_cast<int>(table_size * selectivity);
+      break;
+    case PredicateCondition::GreaterThan:
+    case PredicateCondition::GreaterThanEquals:
+      search_value = static_cast<int>(table_size - table_size * selectivity);
+      break;
+    default:
+      // TODO(cmfcmf) Improve logic for other predicates.
+      Fail("Unsupported predicate condition");
+  }
+  const auto table_wrapper = table_creator(table_size, encoding_type, order_by.value_or(OrderByMode::Ascending));
+  const auto table_column_definitions = table_wrapper->get_output()->column_definitions();
+
+  const auto column_index = ColumnID(0);
+
+  const auto column_definition = table_column_definitions.at(column_index);
+
+  if (column_definition.data_type == DataType::String) {
+    const auto str = std::to_string(type_cast_variant<int>(search_value));
+    search_value = std::string(string_size - str.length(), '0').append(str);
```

Comment:
You are using this method of string padding at at least two other places. I suggest refactoring it.